### ISSUE TITLE
diffing not working (Mac and PC)

### ISSUE THREAD

**Issue by astewart43-gtri** [ubfx/BinDiffHelper#6]
I am having issues with the diffing functionality on both Mac and pc:
MAC:
```
Uncaught Exception! 
java.lang.NullPointerException
	at bindiffhelper.GeneralOpenDialog.okCallback(GeneralOpenDialog.java:101)
	at bindiffhelper.GeneralOpenDialog.lambda$new$0(GeneralOpenDialog.java:86)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2308)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
	at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
	at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:279)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6632)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3342)
	at java.desktop/java.awt.Component.processEvent(Component.java:6397)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5008)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4840)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2772)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4840)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:772)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:743)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:117)
	at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:190)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:235)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:233)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.desktop/java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:233)
	at java.desktop/java.awt.Dialog.show(Dialog.java:1070)
	at java.desktop/java.awt.Component.show(Component.java:1716)
	at java.desktop/java.awt.Component.setVisible(Component.java:1663)
	at java.desktop/java.awt.Window.setVisible(Window.java:1031)
	at java.desktop/java.awt.Dialog.setVisible(Dialog.java:1005)
	at docking.DockingDialog.setVisible(DockingDialog.java:317)
	at docking.DockingWindowManager.lambda$doShowDialog$6(DockingWindowManager.java:1720)
	at ghidra.util.Swing.doRun(Swing.java:295)
	at ghidra.util.Swing.runNow(Swing.java:211)
	at ghidra.util.Swing.runNow(Swing.java:166)
	at ghidra.util.SystemUtilities.runSwingNow(SystemUtilities.java:253)
	at docking.DockingWindowManager.doShowDialog(DockingWindowManager.java:1724)
	at docking.DockingWindowManager.showDialog(DockingWindowManager.java:1770)
	at docking.DockingWindowManager.showDialogOnActiveWindow(DockingWindowManager.java:1651)
	at docking.DockingWindowManager.showDialog(DockingWindowManager.java:1660)
	at bindiffhelper.BinDiffHelperProvider$GeneralOpenAction.actionPerformed(BinDiffHelperProvider.java:461)
	at docking.menu.ToolBarItemManager.lambda$actionPerformed$0(ToolBarItemManager.java:219)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)

```

PC:
```
Uncaught Exception! 
NullPointerException - Cannot invoke "docking.widgets.filechooser.GhidraFileChooserPanel.getFileName()" because "this.extBEFilePanel" is null
java.lang.NullPointerException: Cannot invoke "docking.widgets.filechooser.GhidraFileChooserPanel.getFileName()" because "this.extBEFilePanel" is null
at bindiffhelper.GeneralOpenDialog.okCallback(GeneralOpenDialog.java:101)
at bindiffhelper.GeneralOpenDialog.lambda$new$0(GeneralOpenDialog.java:86)
at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2308)
at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:279)
at java.desktop/java.awt.Component.processMouseEvent(Component.java:6614)
at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3342)
at java.desktop/java.awt.Component.processEvent(Component.java:6379)
at java.desktop/java.awt.Container.processEvent(Container.java:2263)
at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:4990)
at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
at java.desktop/java.awt.Component.dispatchEvent(Component.java:4822)
at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4919)
at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4548)
at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4489)
at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2769)
at java.desktop/java.awt.Component.dispatchEvent(Component.java:4822)
at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:772)
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745)
at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:743)
at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742)
at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:117)
at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:190)
at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:235)
at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:233)
at java.base/java.security.AccessController.doPrivileged(AccessController.java:312)
at java.desktop/java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:233)
at java.desktop/java.awt.Dialog.show(Dialog.java:1070)
at java.desktop/java.awt.Component.show(Component.java:1717)
at java.desktop/java.awt.Component.setVisible(Component.java:1664)
at java.desktop/java.awt.Window.setVisible(Window.java:1028)
at java.desktop/java.awt.Dialog.setVisible(Dialog.java:1005)
at docking.DockingDialog.setVisible(DockingDialog.java:317)
at docking.DockingWindowManager.lambda$doShowDialog$6(DockingWindowManager.java:1720)
at ghidra.util.Swing.doRun(Swing.java:295)
at ghidra.util.Swing.runNow(Swing.java:211)
at ghidra.util.Swing.runNow(Swing.java:166)
at ghidra.util.SystemUtilities.runSwingNow(SystemUtilities.java:253)
at docking.DockingWindowManager.doShowDialog(DockingWindowManager.java:1724)
at docking.DockingWindowManager.showDialog(DockingWindowManager.java:1770)
at docking.DockingWindowManager.showDialogOnActiveWindow(DockingWindowManager.java:1651)
at docking.DockingWindowManager.showDialog(DockingWindowManager.java:1660)
at bindiffhelper.BinDiffHelperProvider$GeneralOpenAction.actionPerformed(BinDiffHelperProvider.java:461)
at docking.menu.ToolBarItemManager.lambda$actionPerformed$0(ToolBarItemManager.java:219)
at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:316)
at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

Ghidra 9.1.2

**Comment by ubfx** [comment]
Thank you for reporting it, I will look into it now

**Comment by astewart43-gtri** [comment]
If you need any other info about my setup, I would be more than happy to provide it.

**Comment by ubfx** [comment]
Are you using the "external BinDiff file" feature? So is this what you see in the dialog?
![image](https://user-images.githubusercontent.com/30509320/96890299-ce58d200-1487-11eb-8a1f-40e6ed655759.png)

I just tried this out with a fresh Windows 10 VM, JDK 11, Ghidra 9.1.2 and BinDiffHelper 0.3 and was able to load a BinDiff file successfully. I get the error you posted when I leave the filename field empty (which will be fixed) but I assume that's not what you're doing

**Comment by astewart43-gtri** [comment]
I was not using the "external BinDiff file" I selected the BinDiff install in the settings page.

**Comment by astewart43-gtri** [comment]
<img width="1196" alt="image" src="https://user-images.githubusercontent.com/62766475/96901984-f5041200-1461-11eb-924e-b61dbf1c1a72.png">

**Comment by ubfx** [comment]
Thank you for the info, I can replicate the problem now and will be posting a fix today, I'll comment here when it is available

**Comment by ubfx** [comment]
Try this version, I think it should fix the problem. It would be nice if you report back if it is indeed fixed, then I'll post the new release:
[ghidra_9.1.2_BinDiffHelper_0.3.1.zip](https://github.com/ubfx/BinDiffHelper/files/5429180/ghidra_9.1.2_BinDiffHelper_0.3.1.zip)

Before installing the new version, please uninstall the old one in Ghidra and remove it from the Ghidra/Extensions folder

**Comment by astewart43-gtri** [comment]
I am no longer getting errors, but it is not hanging on this window: 
<img width="269" alt="image" src="https://user-images.githubusercontent.com/62766475/97007894-191e2c80-1510-11eb-8b08-8ab1b618e2ff.png">

**Comment by ubfx** [comment]
Have both of the project files you are using been analysed?

**Comment by astewart43-gtri** [comment]
they have

**Comment by astewart43-gtri** [comment]
I just tried some much smaller test binaries, C1 and C2 from here: https://github.com/wrongbaud/hackaday-u/tree/master/session-one/exercises
and I got this error
```
Uncaught Exception! 
java.lang.NullPointerException
	at bindiffhelper.BinDiffHelperPlugin.callBinDiff(BinDiffHelperPlugin.java:156)
	at bindiffhelper.GeneralOpenDialog.okCallback(GeneralOpenDialog.java:119)
	at bindiffhelper.GeneralOpenDialog.lambda$new$0(GeneralOpenDialog.java:86)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2308)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
	at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
	at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:279)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6632)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3342)
	at java.desktop/java.awt.Component.processEvent(Component.java:6397)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5008)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4840)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2772)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4840)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:772)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:743)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:117)
	at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:190)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:235)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:233)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.desktop/java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:233)
	at java.desktop/java.awt.Dialog.show(Dialog.java:1070)
	at java.desktop/java.awt.Component.show(Component.java:1716)
	at java.desktop/java.awt.Component.setVisible(Component.java:1663)
	at java.desktop/java.awt.Window.setVisible(Window.java:1031)
	at java.desktop/java.awt.Dialog.setVisible(Dialog.java:1005)
	at docking.DockingDialog.setVisible(DockingDialog.java:317)
	at docking.DockingWindowManager.lambda$doShowDialog$6(DockingWindowManager.java:1720)
	at ghidra.util.Swing.doRun(Swing.java:295)
	at ghidra.util.Swing.runNow(Swing.java:211)
	at ghidra.util.Swing.runNow(Swing.java:166)
	at ghidra.util.SystemUtilities.runSwingNow(SystemUtilities.java:253)
	at docking.DockingWindowManager.doShowDialog(DockingWindowManager.java:1724)
	at docking.DockingWindowManager.showDialog(DockingWindowManager.java:1770)
	at docking.DockingWindowManager.showDialogOnActiveWindow(DockingWindowManager.java:1651)
	at docking.DockingWindowManager.showDialog(DockingWindowManager.java:1660)
	at bindiffhelper.BinDiffHelperProvider$GeneralOpenAction.actionPerformed(BinDiffHelperProvider.java:461)
	at docking.menu.ToolBarItemManager.lambda$actionPerformed$0(ToolBarItemManager.java:219)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```
and now it seems to be hanging here: 
<img width="270" alt="image" src="https://user-images.githubusercontent.com/62766475/97009778-8763ee80-1512-11eb-8350-142bad55688d.png">

**Comment by ubfx** [comment]
You caught lots of bugs there, thanks for reporting! It should all be fixed now, please try version 0.3.2:
https://github.com/ubfx/BinDiffHelper/releases/tag/v0.3.2

**Comment by astewart43-gtri** [comment]
That seems to work! Thank you very much!
